### DISCUSSION TITLE
Tasks that are difficult/painful to follow/understand/solve

### DISCUSSION THREAD

**Discussion by danyaljj** [allenai/natural-instructions#467] [General]
@Palipoor @yeganehkordi While you're addressing human eval feedback #276 let's use this thread to keep track of the tasks that are difficult for humans to understand (**) and we don't have a good way of improving them.  We can collectively discuss ways to improve them or drop them. For completeness, here is [the result of human evaluation with crowdworkers' feedback](https://docs.google.com/spreadsheets/d/1wXStPurP6AamxvglOw0aOJ7V1DiSKczJNyFcyfDnk4w/edit?usp=sharing). 

 (**) not to be confused with tasks that are easy to understand but humans score low in terms of automatic eval.

**Comment by Palipoor** [comment]
Ok!
I didn't know Yeganehis also working on addressing the feedbacks. 
@yeganehkordi Do we need to plan on which task numbers you work on and which I do, so there won't be redundancies?

**Comment by yeganehkordi** [comment]
Sure!
@Palipoor Yeah, I'm working on tasks 200-230 now. To avoid any conflicts, we can work on groups of 100 tasks.

**Reply by Palipoor** [reply]
oh ok! I've already done the 201 - 205. I will jump to 300+ for now.

**Reply by yeganehkordi** [reply]
Thanks!

**Comment by Palipoor** [comment]
Two observations:
- People become overwhelmed when task inputs are long.
- The tasks with "incorrect answer generation" tend to confuse them.

**Reply by Palipoor** [reply]
`task309_race_answer_generation` : It must have been difficult for the annotator. I checked the instances they were given and their gold answer was correct. I don't think the passages were particularly difficult to comprehend. Maybe they simply didn't feel like reading them completely.

**Reply by Palipoor** [reply]
`task311_race_question_generation`: I think the problem with this task is that it doesn't have multiple gold answers, but it certainly can have that. Also, some of the gold outputs seem noisy. + the problem of people not reading carefully.

**Reply by Palipoor** [reply]
The task `task324_jigsaw_classification_disagree` doesn't make sense to me(and the annotators!)

**Reply by Palipoor** [reply]
[!!Warning for swear words below!!]  

In the task `task326_jigsaw_classification_obscene` it seems like the gold labels are "obscene" for sentences containing words like "damn" or "shit", although according to the dictionary definition, and the definition of the task, obscene is something offensive that is somehow related to sex. Should we change the definition and title?

**Reply by Palipoor** [reply]
task 333 has lots of noisy instances!

**Reply by Palipoor** [reply]
task 335 doesn't make sense to me( and the annotators)

**Reply by Palipoor** [reply]
task 345 (part of speech tagging) was difficult for the annotators. (Why do they say "we need to memorize tags?! Aren't the instructions in the same page?)

**Comment by danyaljj** [comment]
Few pieces of intuition from #464: 

> 1. For Winogrande tasks, I think the definitions and the examples were clear. Maybe the tasks themselves are difficult. 
> 2. Tasks 036 and 046 didn't make sense to me, so I couldn't fix them. Either the definition is not good or the instances are too noisy.
> 3. In task 201 all the outputs are "2".

**Comment by yeganehkordi** [comment]
In task 204, based on the [paper](**url**), genres refer to the source of collecting the premises. I don't think if the classification of the sentences based on the genres is practical. For example, two-sided, telephone conversations that took place in 1990 or 1991 (TELEPHONE) and Collection of two-sided, in-person conversations that took place in the early 2000s (FACE-TO-FACE) are hard to distinguish.

**Reply by Palipoor** [reply]
I agree!

**Comment by yeganehkordi** [comment]
Structured Text Generation tasks based on the logic2text (e.g., `task211_logic2text_classification.json`) seem to be difficult for humans who don't have a programming or math background. However, the definitions are good.

**Reply by danyaljj** [reply]
If the definitions are clear (to some human expert) and the instance labels are well-defined, we're good. It's okay if average humans struggle, but there have to be **>0** humans that can solve them.

**Reply by yeganehkordi** [reply]
I think it's okay! Most of the programmers can solve them. Maybe we can use this task in the train split.

**Comment by yeganehkordi** [comment]
In the task `task226_english_language_answer_relevance_classification`, the first half of the instances have the "yes" outputs, and the second half has "no". Maybe we need to shuffle instances of the tasks later.

**Comment by Palipoor** [comment]
task383_matres_classification was difficult for annotators to understand. I didn't understand it either!

**Reply by Palipoor** [reply]
For task "task386_semeval_2018_task3_irony_detection" I had a difficult time understanding what it meant by "irony". The example was "sarcastic" in my opinion. It may be because I am not a native speaker. The annotators seemed to be confused too.

**Comment by Palipoor** [comment]
@yeganehkordi Task 456 was difficult. I think I didn't understand it either.

**Reply by Palipoor** [reply]
Task 471 has very difficult instances, yet the examples are easy.

**Reply by yeganehkordi** [reply]
I'll try to make task 456 more clear.

**Comment by swarooprm** [comment]
@pulkitverma25 wants to help in addressing crowdworker evaluation feedback. What task numbers he can work on, so that it does not conflict with the ones you are working on @Palipoor @yeganehkordi ?

**Reply by swarooprm** [reply]
@aarunku5 also wants to help here, let's decide task numbers for her as well.

**Reply by Palipoor** [reply]
Hi! 
So now Yeganeh has tasks 500 - 600, and I have tasks 600 - 700. So you can split the rest (700+) between the two of you! @aarunku5 @pulkitverma25 
We can also keep a sheet if there're more people involved.

**Reply by pulkitverma25** [reply]
I will start working on tasks 701-800.

**Comment by aarunku5** [comment]
I can do the next 100, i.e., 801-900

**Comment by Palipoor** [comment]
task667_mmmlu_answer_generation_business_ethics seems to be difficult for people.

**Reply by Palipoor** [reply]
The mmmlu tasks are difficult for people. They require field knowledge.

**Comment by yeganehkordi** [comment]
Tasks 268, 274, and 276 were difficult for the annotators. I think the definitions are sufficient.

**Comment by yeganehkordi** [comment]
@Palipoor In task522, here is the question of one of the crowdworkers:
`Can I use parts of sentences to create a whole new sentence? Can I just use one sentence?`
Based on the instances, it seems that they can't combine different sentences. Am I right?

**Reply by Palipoor** [reply]
No, they can't. Unless they're consecutive themselves.

**Reply by yeganehkordi** [reply]
Thanks!

**Comment by Palipoor** [comment]
@aarunku5 @yeganehkordi @pulkitverma25 I will pick up tasks 900 - 1100.

**Reply by Palipoor** [reply]
@aarunku5 @yeganehkordi @pulkitverma25 I will pick up tasks 1100 - 1200.

**Reply by yeganehkordi** [reply]
@aarunku5 @Palipoor @pulkitverma25 I will pick up tasks 1200-1300.

**Reply by Palipoor** [reply]
@yeganehkordi @pulkitverma25 @aarunku5 I will pick up tasks 1300 - 1400.

**Reply by yeganehkordi** [reply]
@aarunku5 @Palipoor @pulkitverma25 I will pick up tasks 1400-1500.

**Comment by yeganehkordi** [comment]
task569_recipe_nlg_text_generation and task573_air_dialogue_classification were difficult for the annotators. Although, the definitions are good.

**Reply by yeganehkordi** [reply]
Task592 was difficult for annotators.

**Reply by danyaljj** [reply]
If you feel the definition are okay (and the task is well-defined), then we're okay.

**Comment by Palipoor** [comment]
task1148_maximum_ascii_value
People didn't know the ascii values and it was difficult for them.

**Reply by Palipoor** [reply]
task1186_nne_hrngo_classification seems to be difficult for people, as the numbers are not very meaningful.

**Comment by pulkitverma25** [comment]
Tasks 664-667, 685-737: 
A common feedback for many tasks is something on these lines: "They should have provided examples with an incorrect selection, rather than just instances that showed answers that didn't accord with the instructions."
Should I add a negative example with an incorrect selection? Or that is not needed?

For now, I created PR#607 without addressing this.

**Reply by Palipoor** [reply]
I think it's good to do this! I also saw some feedbacks on tasks like that.

**Reply by pulkitverma25** [reply]
Ok. This might take time, so I'll create a new PR to address this.

**Comment by Palipoor** [comment]
People seem to be bothered by tasks that may need a google search. Like task1321_country_continent

**Reply by danyaljj** [reply]
Right, I should have specified in the evaluation that they can use Google.

**Comment by CodeHime** [comment]
I will take up changes in the tasks 1500-1600

**Comment by Palipoor** [comment]
I will pick up tasks 1600 - 1700.

**Comment by Palipoor** [comment]
task1625_disfl_qa_asnwer_generation:
A crowdworker has commented:   
`Confusing only because I don't understand why I'd read the disfluent question, when i'd much rather read the proper question and answer it. `
I agree with them!

**Comment by pulkitverma25** [comment]
Picking 1701+

**Comment by pulkitverma25** [comment]
task743_eurlex_summarization seems difficult for annotators (and me).
One feedback:
"writing articles for the headline requires knowledge about anything legal related and I don't have that knowledge"

I agree with the feedback. This is not something you can Google or answer using your knowledge or inference unless you have "legal" knowledge.

**Reply by pulkitverma25** [reply]
Same feedback for task744_eurlex_classification
Too much text to read, which is also full of "legal" vocabulary.
We can maybe mention that good knowledge of this kind is required for this task.

**Reply by pulkitverma25** [reply]
@danyaljj Do you think eurlex tasks (743 and 744) make sense for the users? The passages are long and need knowledge of legal terminology to answer. And their answers can't be Googled too.
